thirdparty/hnswlib/hnswlib/hnswalg.h

```diff
@@ -1240,11 +1241,16 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
 
         // do bruteforce search when delete rate high
         if (!bitset.empty()) {
-            if (bitset.count() >= (cur_element_count * kHnswSearchKnnBFThreshold)) {
+            if (bitset.count() >= (cur_element_count * kHnswSearchKnnBFFilterThreshold)) {
                 return searchKnnBF(query_data, k, bitset);
             }
         }
 
+        // do bruteforce search when topk is super large
+        if (k >= (cur_element_count * kHnswSearchBFTopkThreshold)) {
```

Comment:
The definition of `big top_k` scenario  seems also need take the `filter_ratio` into consideration. For example, if filter_ratio is 60%, then `top_k = cur_element_count  * 40%` is big enough because all unfiltered data will be selected,  and graph searching may return missing result for nodes not fully connected. However, more tests should be added. Hotfix now?

Reply:
Good point! Updated